tests/Rule/EnsureExplicitNullableTypesTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of DOCtor-RST.
+ *
+ * (c) Oskar Stark <oskarstark@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Rule;
+
+use App\Rule\EnsureExplicitNullableTypes;
+use App\Tests\RstSample;
+use App\Tests\UnitTestCase;
+use App\Value\NullViolation;
+use App\Value\Violation;
+use App\Value\ViolationInterface;
+
+final class EnsureExplicitNullableTypesTest extends UnitTestCase
+{
+    /**
+     * @test
+     *
+     * @dataProvider invalidProvider
+     * @dataProvider validProvider
+     */
+    public function check(ViolationInterface $expected, RstSample $sample): void
+    {
+        self::assertEquals(
+            $expected,
+            (new EnsureExplicitNullableTypes())->check($sample->lines, $sample->lineNumber, 'filename'),
+        );
+    }
+
+    /**
+     * @return \Generator<string, array{0: ViolationInterface, 1: RstSample}>
+     */
+    public static function validProvider(): iterable
+    {
+        $validCases = [
+            'function foo(?int $bar = null)',
+            'function foo(int|null $bar = null)',
```

Comment:
We should add cases with multiple parameters, and according to the regex, multiple or no spaces

`'function foo(int|null $bar = null, int $baz = null)'` is considered valid but is not

We can also add `'function foo($payload = null)'` as valid

Reply:
I would only handle Parma's with types

Reply:
```function foo($payload = null)``` is something we maybe don't want to "promote", what do you think ? 

In all manner, it won't raise violation here

---

`function foo(int|null $bar = null, int $baz = null)`

Thanks i'll update the code (maybe could we add a matchAll() method there ?)

Reply:
Oh nevermind, i've read the code 😮‍💨